dspace-api/src/main/java/org/dspace/identifier/FilteredIdentifierProvider.java

```diff
@@ -24,61 +24,63 @@
  */
 public abstract class FilteredIdentifierProvider extends IdentifierProvider {
 
-    protected Filter filterService;
+    protected Filter filter = new TrueFilter();
 
     /**
-     * Setter for spring to set the filter service from the property in configuration XML
-     * @param filterService - an object implementing the org.dspace.content.logic.Filter interface
+     * Setter for spring to set the default filter from the property in configuration XML
+     * @param filter - an object implementing the org.dspace.content.logic.Filter interface
      */
-    @Autowired
-    public void setFilterService(Filter filterService) {
-        this.filterService = filterService;
+    public void setFilter(Filter filter) {
+        this.filter = filter;
```

Comment:
Minor: since TrueFilter is assigned by default, I would change the code here. I would reassign it if and olny if filter parameter is not null, do something like

Optional.ofNullable(filter).map(f -> this.filter = f)

Behavior should not change

Reply:
This method did have a null check, but I've updated it to your one-liner as it seems more elegant.

Reply:
update: this does't compile and I wasn't fully certain of the pattern.
So i replaced it with a simple null check that is better than my original, but also easy to follow and doesn't fail build